src/app/modules/entities/state-entity-list/state-entity-list.component.ts

```diff
@@ -0,0 +1,99 @@
+import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
+import { MatPaginator, MatTableDataSource, PageEvent, MatDialog } from '@angular/material';
+import { EntityService } from '../../admin-core/services/entity-service/entity.service';
+import { CommonServiceService } from '../../admin-core';
+import { Router } from '@angular/router';
+
+
+@Component({
+  selector: 'app-state-entity-list',
+  templateUrl: './state-entity-list.component.html',
+  styleUrls: ['./state-entity-list.component.scss']
+})
+export class StateEntityListComponent implements OnInit {
+  EntityStateList: any;
+  dataSource: MatTableDataSource<any>;
+  displayedColumns: any = [];
+  listing: boolean = false;
+  recordCount: any;
+  columns: any;
+  fieldsForOrganisation: any;
+  formdata: any;
+  confirmPopupResult: any;
+  status: any = '';
+  orgObject: any;
+  assignedStatus: any;
+  search: any;
+  noData: boolean;
+  queryParams = {
+    page: 1,
+    size: 10,
+  };
+  @ViewChild('searchInput') searchInput: ElementRef;
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+
+  constructor(private entityService: EntityService,
+    private commonServiceService: CommonServiceService,
+    private router: Router) { }
+
+  ngOnInit() {
+    this.getEntityStateList();
+  }
+
+  applyFilter(event: Event) {
+    const filterValue = (event.target as HTMLInputElement).value;
+    this.dataSource.filter = filterValue.trim().toLowerCase();
+    if (this.dataSource.filteredData.length < 1) {
+      this.noData = true;
+    } else {
+      this.noData = false;
+    }
+
+
+    if (this.dataSource.paginator) {
+      this.dataSource.paginator.firstPage();
+    }
+  }
+
+  /**
+  * To get EntityStateList
+  */
+  getEntityStateList() {
+    this.entityService.getStatesList(this.queryParams).subscribe(data => {
+      console.log('getEntityStateList', data);
+      this.EntityStateList = data['result'];
+      // this.refreshDatasource(data['result']['data']);
+      this.displayedColumns = [];
+      this.dataSource = new MatTableDataSource(data['result'].data);
+      this.columns = data['result']['columns'];
+      if (this.EntityStateList) {
+        this.columns.forEach(element => {
+          if (element.visible) {
+            this.displayedColumns.push(element.key)
+          }
+        });
+      }
+      this.recordCount = data['result'].count;
+      this.listing = true;
+    }, error => {
+      this.listing = true;
+      this.commonServiceService.errorHandling(error);
+    });
+  }
+
+  // get color based on the status
+  getItemCssClassByStatus(status): string {
+    switch (status) {
+      case 'Active':
+        return 'active';
+      case 'Inactive':
+        return 'inactive';
+    }
+    return '';
```

Comment:
why we need this return

Reply:
we are returning the class when other than this we are passing empty

Reply:
there is a default keyword in switch case,,. Please use that

Reply:
Removed that empty also